File: xtcdata/xtcdata/xtc/src/TransitionId.cc
#include "xtcdata/xtc/TransitionId.hh"

using namespace XtcData;

const char* TransitionId::name(TransitionId::Value id)
{
    static const char* _names[] = {
        "ClearReadout",
        "Reset",
        "Configure",
        "Unconfigure",
        "BeginRun",
        "EndRun",
        "BeginStep",
        "EndStep",
        "Enable",
        "Disable",
        "SlowUpdate",
        "Unused_11",
        "L1Accept",
    };
    return (id < TransitionId::NumberOf ? _names[id] : "-Invalid-");

    // Bail on compilation if someone forgets to update this list
    static_assert(sizeof(_names) / sizeof(*_names) == TransitionId::NumberOf,"test message");
};

File: psana/psana/psexp/TransitionId.py
ClearReadout = 0
Reset = 1
Configure = 2
Unconfigure = 3
BeginRun = 4
EndRun = 5
BeginStep = 6
EndStep = 7
Enable = 8
Disable = 9
SlowUpdate = 10
L1Accept_EndOfBatch = 11
L1Accept = 12
NumberOf = 13


def isEvent(transition_id):
    if transition_id == L1Accept or transition_id == L1Accept_EndOfBatch:
        return True
    return False
